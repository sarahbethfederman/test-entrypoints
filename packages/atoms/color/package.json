{
  "name": "@auscred/color",
  "version": "0.0.1",
  "source": "src/Color.tsx",
  "main": "dist/Colour.js",
  "types": "dist/Colour.d.js",
  "files": [
    "lib"
  ],
  "keywords": [],
  "author": "Lendi Devs",
  "license": "ISC",
  "peerDependencies": {
    "@types/react": "^16.4.0",
    "react": "^16.4.0",
    "styled-components": "^3.3.3"
  },
  "scripts": {
    "build": "tsc --pretty"
  }
}
  