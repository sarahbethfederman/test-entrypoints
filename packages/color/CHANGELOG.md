2.0.2 - New Changelog
