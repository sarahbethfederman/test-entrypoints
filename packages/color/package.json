{
  "name": "@lendi-ui/color",
  "version": "3.0.0",
  "description": "Color utilities for color.",
  "source": "src/index.ts",
  "main": "dist/cjs/index.js",
  "module": "dist/esm/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist"
  ],
  "license": "ISC",
  "peerDependencies": {
    "@lendi-ui/theme": "^6.0.0",
    "styled-components": "^3"
  },
  "dependencies": {
    "@lendi-ui/breakpoint": "^2.0.0"
  },
  "scripts": {
    "build:esm": "tsc --project tsconfig.esm.json",
    "build:types": "tsc --project tsconfig.types.json",
    "build": "yarn run build:types && yarn run build:esm"
  }
}
