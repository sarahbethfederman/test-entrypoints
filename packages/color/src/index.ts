import { css } from 'styled-components';
import { select } from '@lendi-ui/theme';
import { map, BreakpointValue, BreakpointValueMap } from '@lendi-ui/breakpoint';

export type NameOrNameMap = BreakpointValue<string> | BreakpointValueMap<string>;

export function color(name: NameOrNameMap) {
  return select(`colors.${name}`);
}

export function fg(name: NameOrNameMap) {
  return map(
    name,
    (n?: string) =>
      n === undefined
        ? ''
        : css`
            color: ${color(n)};
          `
  );
}

export function bg(name: NameOrNameMap) {
  return map(
    name,
    (n?: string) =>
      n === undefined
        ? ''
        : css`
            background-color: ${color(n)};
          `
  );
}
