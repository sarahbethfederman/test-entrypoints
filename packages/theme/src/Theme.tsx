import * as React from 'react';
import styled, { ThemeProvider, createGlobalStyle, GlobalStyleComponent, DefaultTheme } from 'styled-components';
import { Breakpoint } from '@lendi-ui/breakpoint';
import { theme } from './theme/lendi';
import { ThemeMap } from './types';

interface ThemeProps {
  kind?: ThemeMap;
  children?: React.ReactNode;
}

// Any cross browser resets should occur here
const GlobalStyle: GlobalStyleComponent<{ theme: ThemeMap }, DefaultTheme> = createGlobalStyle`
  html, body {
    margin: 0;
    padding: 0;
    font-family: ${({ theme }) => theme.typography.body};
  }

  h1, h2, h3, h4, h5, h6 {
    font-family: ${({ theme }) => theme.typography.heading};
    font-weight: 700;
  }

  input::-ms-clear {
    display: none;
  }

  :root {
    --lendi-ui-size: 14px;
  }

  @media (min-width: ${Breakpoint.md}) {
    :root {
      --lendi-ui-size: 16px;
    }
  }
`;

const Root = styled.div`
  color: ${theme.colors['shade.700']};
`;

export function Theme(props: ThemeProps) {
  const { children } = props;
  const kind = props.kind || theme;

  return (
    <ThemeProvider theme={kind}>
      <Root>
        {children}
        <GlobalStyle />
      </Root>
    </ThemeProvider>
  );
}
