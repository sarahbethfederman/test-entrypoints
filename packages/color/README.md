# @lendi-ui/color

Color style with `bg`(background color), `fg`(font color) and `color`(every CSS property relate to color).

## Installation

```
yarn add @lendi-ui/color
```

## Usage

- `bg` for styling background-color.

```jsx
import { bg } from '@lendi-ui/color';
const Square = styled.div`
  width: 2em;
  height: 2em;
  ${bg('warn.500')};
`;
```

- `fg` for styling font-color.

```jsx
import { fg } from '@lendi-ui/color';
const WarningText = styled.span`
  ${fg('warn.500')};
`;
```

- `color` for styling every CSS porperty relate to color.

```jsx
import { color } from '@lendi-ui/color';
const Circle = styled.div`
  width: 0;
  border-radius: 1em;
  border: 1em solid ${color('primary.500')};
`;
```
