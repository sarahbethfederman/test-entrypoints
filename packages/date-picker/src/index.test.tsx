import * as React from 'react';
import { mount } from 'enzyme';
import Theme from '@lendi-ui/theme';
import color from '@lendi-ui/color';
import DatePicker from './index';
import {
  Wrapper,
  IconWrapper,
  InputWrapper,
  DayInputWrapper,
  MonthInputWrapper,
  YearInputWrapper,
  DatePickerSize,
} from './index.style';

const sizes: DatePickerSize[] = ['lg', 'md', 'sm'];

let element;

function render(props) {
  element = mount(
    <Theme>
      <DatePicker dayOnChange={jest.fn()} monthOnChange={jest.fn()} yearOnChange={jest.fn()} {...props} />
    </Theme>
  );
}

describe('DatePicker', () => {
  it('should render the component with month and year sections', () => {
    render({});

    expect(element.find(DatePicker)).toHaveLength(1);
    expect(element.find(DatePicker)).toMatchSnapshot();
    expect(element.find(Wrapper)).toHaveLength(1);
    expect(element.find(Wrapper)).not.toHaveStyleRule('width', '100%');
    expect(element.find(IconWrapper)).toHaveLength(1);
    expect(element.find(DayInputWrapper)).toHaveLength(0);
    expect(element.find(MonthInputWrapper)).toHaveLength(1);
    expect(element.find(YearInputWrapper)).toHaveLength(1);
    expect(element.find(InputWrapper)).toHaveLength(2);
  });

  describe('the hasDayField conditional', () => {
    it('should render the day section when hasDayField is true', () => {
      render({ hasDayField: true });

      expect(element.find(DayInputWrapper)).toHaveLength(1);
      expect(element.find(InputWrapper)).toHaveLength(3);
    });

    it('should not render the day section when hasDayField is false', () => {
      render({ hasDayField: false });

      expect(element.find(DayInputWrapper)).toHaveLength(0);
      expect(element.find(InputWrapper)).toHaveLength(2);
    });

    it('should not render the day section when no hasDayField prop is given', () => {
      render({});

      expect(element.find(DayInputWrapper)).toHaveLength(0);
      expect(element.find(InputWrapper)).toHaveLength(2);
    });
  });

  describe('the isDisabled conditional', () => {
    it('should render with readonly style when isDisabled is true', () => {
      render({ isDisabled: true });

      element.find(InputWrapper).forEach((inputWrapper) => {
        expect(inputWrapper.prop('readOnly')).toEqual(true);
      });
    });

    it('should not render with readonly style when isDisabled is false', () => {
      render({ isDisabled: false });

      element.find(InputWrapper).forEach((inputWrapper) => {
        expect(inputWrapper.prop('readOnly')).toEqual(false);
      });
    });

    it('should not render with readonly style when no isDisabled is given', () => {
      render({});

      element.find(InputWrapper).forEach((inputWrapper) => {
        expect(inputWrapper.prop('readOnly')).toEqual(false);
      });
    });
  });

  describe('it should match all styles in different size of the DatePicker component', () => {
    sizes.forEach((size) => {
      describe(`${size}`, () => {
        it('should render with the day field in the DatePicker component', () => {
          render({ hasDayField: true });

          expect(element.find(DayInputWrapper)).toHaveLength(1);
          expect(element.find(InputWrapper)).toHaveLength(3);
        });

        it('should render readonly style of the DatePicker component', () => {
          render({ isDisabled: true });
          element.find(InputWrapper).forEach((inputWrapper) => {
            expect(inputWrapper.prop('readOnly')).toEqual(true);
          });
        });

        it('should render error style of the DatePicker component', () => {
          render({ isError: true });

          expect(element.find(Wrapper).prop('isError')).toEqual(true);
        });
      });
    });
  });

  describe('test native props and Standard HTML Attributes', () => {
    it('should mount with Aria attributes', () => {
      const ARIA_LABEL = 'testLabel';
      const ARIA_DESCRIBE_BY = 'info';
      render({ 'aria-label': ARIA_LABEL, 'aria-describedby': ARIA_DESCRIBE_BY });
      const datePickerAttributes = element.find(DatePicker).props();
      expect(datePickerAttributes['aria-label']).toBe(ARIA_LABEL);
      expect(datePickerAttributes['aria-describedby']).toBe(ARIA_DESCRIBE_BY);
    });
    it('should mount with native props like id, className', () => {
      const TEXT_ID = 'testId';
      const CLASS_NAME = 'testClass';
      render({
        id: TEXT_ID,
        className: CLASS_NAME,
      });
      const datePickerAttributes = element.find(DatePicker).props();
      expect(datePickerAttributes.id).toBe(TEXT_ID);
      expect(datePickerAttributes.className).toBe(CLASS_NAME);
    });
  });
});
