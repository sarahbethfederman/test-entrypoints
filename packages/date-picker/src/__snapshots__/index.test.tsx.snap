// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker should render the component with month and year sections 1`] = `
.c2 {
  width: 1em;
  height: 1em;
  vertical-align: middle;
  display: inline-block;
}

.c2 > svg {
  fill: rgba(0,192,165,1);
  width: inherit;
  height: inherit;
}

.c2 .fillPath {
  fill: rgba(0,192,165,1);
}

.c0 {
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  background-color: rgba(255,255,255,1);
  height: calc(3 * var(--lendi-ui-size));
  font-size: calc(1.125 * var(--lendi-ui-size));
  min-width: calc(11.75 * var(--lendi-ui-size));
  border: 1px solid rgba(200,200,200,1);
  font-family: 'Open Sans',sans-serif;
  border-radius: 6px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c0:hover {
  border: solid 1px rgba(95,95,95,1);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: calc(3 * var(--lendi-ui-size));
  padding-top: calc(0.1875 * var(--lendi-ui-size));
  font-size: calc(1.875 * var(--lendi-ui-size));
}

.c4 {
  font-size: inherit;
  height: 100%;
  width: 100%;
  border: 0px;
  padding: 0px;
  text-align: center;
}

.c6 {
  border-bottom-right-radius: 6px;
  border-top-right-radius: 6px;
  font-size: inherit;
  height: 100%;
  width: 100%;
  border: 0px;
  padding: 0px;
  text-align: center;
}

.c3 {
  border-left: solid 1px rgba(200,200,200,1);
  width: 100%;
  height: 100%;
}

.c5 {
  border-bottom-right-radius: 6px;
  border-top-right-radius: 6px;
  border-left: solid 1px rgba(200,200,200,1);
  width: 100%;
  height: 100%;
}

<DatePicker
  dayOnChange={[MockFunction]}
  monthOnChange={[MockFunction]}
  yearOnChange={[MockFunction]}
>
  <styled.div
    hasDayField={false}
    isDisabled={false}
    isError={false}
    size="md"
  >
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-htpNat",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              "
  box-sizing: border-box;
  line-height: 1.333;
  white-space: normal;
  border-spacing: 2px;
  text-align: left;
  word-spacing: normal;
  letter-spacing: normal;
",
              "
  ",
              "
          background-color: ",
              [Function],
              ";
        ",
              "
  ",
              [Function],
              ";
  ",
              [Function],
              ";
  ",
              [Function],
              "
  font-family: ",
              [Function],
              ";
  border-radius: ",
              [Function],
              ";
  display: flex;
  flex-direction: row;
  align-items: flex-start;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-htpNat",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      hasDayField={false}
      isDisabled={false}
      isError={false}
      size="md"
    >
      <div
        className="c0"
        size="md"
      >
        <styled.div
          size="md"
        >
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: center;
  ",
                    [Function],
                    ";
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
            size="md"
          >
            <div
              className="c1"
              size="md"
            >
              <Calendar
                color="primary.500"
              >
                <styled.span
                  color="primary.500"
                >
                  <StyledComponent
                    color="primary.500"
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-bwzfXH",
                          "isStatic": false,
                          "lastClassName": "c2",
                          "rules": Array [
                            "
  width: ",
                            [Function],
                            ";
  height: ",
                            [Function],
                            ";
  vertical-align: middle;
  display: inline-block;

  & > svg {
    fill: ",
                            [Function],
                            ";
    width: inherit;
    height: inherit;
  }

  .fillPath {
    fill: ",
                            [Function],
                            ";
  }
",
                          ],
                        },
                        "displayName": "styled.span",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-bwzfXH",
                        "target": "span",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <span
                      className="c2"
                      color="primary.500"
                    >
                      <svg
                        fill="currentColor"
                        height="1em"
                        viewBox="0 0 24 24"
                        width="1em"
                      >
                        <path
                          d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19a2 2 0 002 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                        />
                      </svg>
                    </span>
                  </StyledComponent>
                </styled.span>
              </Calendar>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bZQynM",
                  "isStatic": false,
                  "lastClassName": "c3",
                  "rules": Array [
                    "
  border-left: solid 1px ",
                    [Function],
                    ";
  width: 100%;
  height: 100%;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bZQynM",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c3"
            >
              <styled.input
                maxLength={2}
                onChange={[Function]}
                placeholder="MM"
                readOnly={false}
                type="text"
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-ifAKCX",
                        "isStatic": false,
                        "lastClassName": "c6",
                        "rules": Array [
                          "
  ",
                          [Function],
                          ";

  ",
                          [Function],
                          "
  font-size: inherit;
  height: 100%;
  width: 100%;
  border: 0px;
  padding: 0px;
  text-align: center;
",
                        ],
                      },
                      "displayName": "styled.input",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-ifAKCX",
                      "target": "input",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  maxLength={2}
                  onChange={[Function]}
                  placeholder="MM"
                  readOnly={false}
                  type="text"
                >
                  <input
                    className="c4"
                    maxLength={2}
                    onChange={[Function]}
                    placeholder="MM"
                    readOnly={false}
                    type="text"
                  />
                </StyledComponent>
              </styled.input>
            </div>
          </StyledComponent>
        </styled.div>
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-gzVnrw",
                  "isStatic": false,
                  "lastClassName": "c5",
                  "rules": Array [
                    "
  border-bottom-right-radius: 6px;
  border-top-right-radius: 6px;
  border-left: solid 1px ",
                    [Function],
                    ";
  width: 100%;
  height: 100%;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-gzVnrw",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c5"
            >
              <styled.input
                isRightEdge={true}
                maxLength={4}
                onChange={[Function]}
                placeholder="YYYY"
                readOnly={false}
                type="text"
              >
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-ifAKCX",
                        "isStatic": false,
                        "lastClassName": "c6",
                        "rules": Array [
                          "
  ",
                          [Function],
                          ";

  ",
                          [Function],
                          "
  font-size: inherit;
  height: 100%;
  width: 100%;
  border: 0px;
  padding: 0px;
  text-align: center;
",
                        ],
                      },
                      "displayName": "styled.input",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-ifAKCX",
                      "target": "input",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                  isRightEdge={true}
                  maxLength={4}
                  onChange={[Function]}
                  placeholder="YYYY"
                  readOnly={false}
                  type="text"
                >
                  <input
                    className="c6"
                    maxLength={4}
                    onChange={[Function]}
                    placeholder="YYYY"
                    readOnly={false}
                    type="text"
                  />
                </StyledComponent>
              </styled.input>
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</DatePicker>
`;
